### ISSUE TITLE
/etc/transmission/updatePort.sh seems to not work properly

### ISSUE THREAD

**Issue by a-lagopus** [haugene/docker-transmission-openvpn#78]
Hi, I've been using your script for a while and it's really useful.
However it seems that the /etc/transmission/updatePort.sh is not working properly...
In the logs, I see that the port isn't really updated
{"log":"Got new port 32069 from pia\n","stream":"stdout","time":"2016-05-15T13:35:54.527651501Z"}
{"log":"Checking port...\n","stream":"stdout","time":"2016-05-15T13:35:54.569254009Z"}
{"log":"Got new port 32069 from pia\n","stream":"stdout","time":"2016-05-15T14:11:36.122131619Z"}
{"log":"Checking port...\n","stream":"stdout","time":"2016-05-15T14:11:36.169260958Z"}

The script uses a hyperlink 
https://www.privateinternetaccess.com/vpninfo/port_forward_assignment, but when I try to follow it, I see a "page not found" error.
I can only speculate that it was removed or something.
Also after the script has fired my router sometimes hangs and needs to be restarted - I still couldn't track it in the logs, though...

I'm quite new with the whole git thing, so please excuse me if I don't provide enough information.

**Comment by sylarevan** [comment]
The URL used in the script seems correct to me, and is identical to the one used in the script proposed in PIA forums. Nevetheless, it seems that par forwarding is not really working well, according to [this](https://www.privateinternetaccess.com/forum/discussion/3359/port-forwarding-without-application-pia-script-advanced-users/p4).
BTW, where can I find these logs related to port updating, or even openvpn logs?

**Comment by a-lagopus** [comment]
I generally look in the /var/lib/docker/containers/{yourcontainerfullname}/{yourcontainerfullname}.log
Where the full container name is from docker ps -a

So looks like people get it working only after some heavy tinkering, just calling the URL doesn't work anymore.

**Comment by sylarevan** [comment]
FYI, here is what I get with docker logs (a LOT of times):

> 2016-05-22T13:21:45.076990869Z   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2016-05-22T13:21:45.077752533Z                                  Dload  Upload   Total   Spent    Left  Speed
100   152  100    60  100    92    224    343 --:--:-- --:--:-- --:--:--   343
2016-05-22T13:21:45.348658235Z Got new port  from pia
2016-05-22T13:21:45.350919847Z transmission auth required
2016-05-22T13:21:45.396230425Z Unexpected response: <h1>401: Unauthorized</h1>Unauthorized UserwMGMxd1U0SktjSA==
2016-05-22T13:21:45.396428285Z User-Agent: transmission-remote/2.92 (14714)
2016-05-22T13:21:45.396525260Z Host: localhost:9091
2016-05-22T13:21:45.396608991Z Accept: */*
2016-05-22T13:21:45.396695460Z Accept-Encoding: deflate, gzip
2016-05-22T13:21:45.396780154Z
2016-05-22T13:21:45.397630030Z No action needed, port hasn't changed

So it seems I get a new port from the script, but setting it in transmission seems to fail with a 401 error (I have set authentication on in transmission).

**Comment by haugene** [comment]
Hmm. I just pulled the latest image and I am using username/password for transmission. But I use quite a simple one, can this be encoding? Are you using special characters? Try setting username and password to asd/asd or something and see if it works.

The update script will log to stdout as the other processes. You can se the container log by typing `docker logs <container name/id>`. Paste the logs here if it doesn't work.

**Comment by a-lagopus** [comment]
Hi, thanks for your reply!
I didn't know about the docker logs command, it's very useful.
However even after I changed the transmission pass/login to simple ones, the problem remains - pia doesn't return a new random port - it's the same one every time.

I am attaching the docker log file from the container.
It's running on Ubuntu Server LTS x64, connected to the internet via router with UPnP and port forwarding disabled.
I'm not sure but I think my router dropping connection is actually a separate issue.

[docker_log.txt](https://github.com/haugene/docker-transmission-openvpn/files/288028/docker_log.txt)

Thank you very much for your help!

**Comment by haugene** [comment]
This is the correct behavior. On first run a random client id is generated and sent to pia for each request to ensure that we do. We get an assigned port and check back with pia every hour to see that this still is the port the have opened for us. Then it prints that no change is needed as transmission already has set this port.

**Comment by a-lagopus** [comment]
Oh, sorry I misunderstood.
I was quite sure that the forum thread that mentioned the link used in a script https://www.privateinternetaccess.com/vpninfo/port_forward_assignment talked about getting a random port every hour or so to maximize safety and "for nobody to be able to guess your port" or something like that.
Back to fixing my router problem then - still can't determine whether it's related to the container activity or not.
Thank you for your help!

**Comment by haugene** [comment]
No worries :)

As for getting a new port every hour, I don't know. Could be that it increases security somehow, but then again you are announcing this port with the tracker of all of your torrents so it isn't really a secret.
It would be a kind of security by obscurity in that case. The actual security here is implemented by Transmisison on the listening port.

Good luck :) The container activity should not be related to your router problem I hope. The container is just another internet device, not requiring any special settings. Let me know if it is though!
